spark/src/main/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuery.scala

```diff
@@ -60,37 +127,199 @@ trait OptimizeMetadataOnlyDeltaQuery {
+          // should be equals to numFiles
+          Seq(
+            s"""case when $numFiles = count(`nullCount.$physicalName`)
+               | AND $numFiles = (count(`min.$physicalName`) + sum(case when
+               |  ISNULL(`min.$physicalName`) and `nullCount.$physicalName` = numRecords
+               |   then 1 else 0 end))
+               | AND $numFiles = (count(`max.$physicalName`) + sum(case when
+               |  ISNULL(`max.$physicalName`) AND `nullCount.$physicalName` = numRecords
+               |   then 1 else 0 end))
+               | then TRUE else FALSE end as `complete_$physicalName`""".stripMargin,
+            s"min(`min.$physicalName`) as `min_$physicalName`",
+            s"max(`max.$physicalName`) as `max_$physicalName`")
+        }
+
+        val statsResults = files.select(columnsToQuery: _*).selectExpr(minMaxExpr: _*).head
+
+        dataColumnsWithStats
+          .filter(x => statsResults.getAs[Boolean](s"complete_${x._2}"))
+          .map { columnAndPhysicalName =>
+            val column = columnAndPhysicalName._1
+            val physicalName = columnAndPhysicalName._2
+            column.name ->
+              DeltaColumnStat(
+                statsResults.getAs(s"min_$physicalName"),
+                statsResults.getAs(s"max_$physicalName"))
+          }.toMap
+      }
+    }
+
+    def extractGlobalPartitionedColumnStatsDeltaLog(snapshot: Snapshot):
+    Map[String, DeltaColumnStat] = {
+
+      val partitionedColumns = snapshot.metadata.partitionSchema
+        .filter(x => columnStatsSupportedDataTypes.contains(x.dataType))
+        .map(x => (x, DeltaColumnMapping.getPhysicalName(x)))
+
+      if (partitionedColumns.isEmpty) {
+        Map.empty
+      } else {
+        val partitionedColumnsValues = partitionedColumns.map { partitionedColumn =>
+          val physicalName = partitionedColumn._2
+          col(s"partitionValues.`$physicalName`")
+            .cast(partitionedColumn._1.dataType).as(physicalName)
+        }
+
+        val partitionedColumnsAgg = partitionedColumns.flatMap { partitionedColumn =>
+          val physicalName = partitionedColumn._2
+
+          Seq(min(s"`$physicalName`").as(s"min_$physicalName"),
+            max(s"`$physicalName`").as(s"max_$physicalName"))
+        }
+
+        val partitionedColumnsQuery = snapshot.allFiles
+          .select(partitionedColumnsValues: _*)
+          .agg(partitionedColumnsAgg.head, partitionedColumnsAgg.tail: _*)
+          .head()
+
+        partitionedColumns.map { partitionedColumn =>
+          val physicalName = partitionedColumn._2
+
+          partitionedColumn._1.name ->
+            DeltaColumnStat(
+              partitionedColumnsQuery.getAs(s"min_$physicalName"),
+              partitionedColumnsQuery.getAs(s"max_$physicalName"))
+        }.toMap
+      }
     }
+
+    CaseInsensitiveMap(
+      extractGlobalColumnStatsDeltaLog(snapshot).++
+      (extractGlobalPartitionedColumnStatsDeltaLog(snapshot)))
```

Comment:
do we need to extract the partition column stats always? Can this be optional based on whether there is agg expr on partition column?